esmvalcore/preprocessor/_mask.py

```diff
@@ -224,12 +247,12 @@ def mask_landseaice(cube, fx_files, mask_out):
     return cube
 
 
-def _get_geometry_from_shp(shapefilename):
+def _get_geometries_from_shp(shapefilename):
     """Get the mask geometry out from a shapefile."""
     reader = shpreader.Reader(shapefilename)
-    # Index 0 grabs the lowest resolution mask (no zoom)
-    main_geom = [contour for contour in reader.geometries()][0]
-    return main_geom
+    geometries = [contour for contour in reader.geometries()]
```

Comment:
@bascrezee try using `fiona`:
```
with fiona.open(shapefilename) as geometries:
    for item in geometries:
        shape = shapely.geometry.shape(item['geometry'])
```
see example in `preprocessor/_area.py/extract_shape()`
Also be aware of the issue that (most of the times) the shapefiles go from -180 to +180, see solution here https://github.com/ESMValGroup/ESMValCore/pull/319

Reply:
Thanks for the reviewing, can you also have a look at my comment, above. Basically I suggest to cancel this PR first, and restructure the code before I add this new mask again.

Reply:
I think also when switching from `cartopy` to `fiona` for reading the shapefiles, this should be implemented before. I simply followed examples for the other masks.

Reply:
No need to switch, since `cartopy` actually uses `fiona`? https://github.com/SciTools/cartopy/blob/0e5eef86afe3502bfc41b84f71ba2bf48e2812bd/lib/cartopy/io/shapereader.py#L54

Reply:
>  I think it is best if I close this PR and that you/we start rewriting the existing functions in such a way that new masks can be added later without touching any of the existing masks. Please look at the changes in this draft PR and let me know if you get my point :)

ping @valeriupredoi

Reply:
@bascrezee have a look at #389 - you can plug in all the regions you want and not change the existinmg mask landsea :beer: